stdlib/public/runtime/RuntimeInvocationsTracking.h

```diff
@@ -0,0 +1,110 @@
+//===--- RuntimeInvocationsTracking.h ---------------------------*- C++ -*-===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2017 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+//
+// Track invocations of Swift runtime functions. This can be used for
+// performance analysis.
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef SWIFT_STDLIB_RUNTIME_INVOCATIONS_TRACKING_H
+#define SWIFT_STDLIB_RUNTIME_INVOCATIONS_TRACKING_H
+
+namespace swift {
+struct HeapObject;
+struct RuntimeFunctionCountersState;
+} // end namespace swift
+
+/// The name of a helper function for tracking the calls of a runtime function.
+#define SWIFT_RT_TRACK_INVOCATION_NAME(RT_FUNCTION)                            \
```

Comment:
Why is this outside the #if !defined(NDEBUG)

Reply:
Because it is used also by the RuntimeInvocationsTracking.cpp, which is compiled always, even if NDEBUG is defined.